avocado/core/job.py

```diff
@@ -275,6 +275,9 @@ def _make_test_suite(self, references=None):
         loader.loader.load_plugins(self.args)
         try:
             suite = loader.loader.discover(references)
```

Comment:
I remember talking about adding the tags info to the test factories, and doing the filtering (optionally) later on.  I chose to keep the filtering out of the `discover()` for a few reasons:
 * API changes would be necessary 
 * I'm not confident we should add a "tags" parameter to loader at this point as that the loader API is one of the remaining ones to be turned into the "plugin interfaces" model
 * The two phase (discover/filter) is a lot easier to understand and debug
 * I don't think (but I lack numbers) that it impacts negatively the performance (it's only called when the filter arguments are given)